### DISCUSSION TITLE
Option to delete failed downloads

### DISCUSSION THREAD

**Discussion by alexjshaw** [Emy69/CoomerDL#33] [Ideas]
First things first, I love the program.

The one issue I have is that coomer is quite unreliable, which results in files that often throw this kind of error:

Error downloading: ('Connection broken: IncompleteRead(100663296 bytes read, 200581899 more expected)', IncompleteRead(100663296 bytes read, 200581899 more expected))

and you end up with a file that is only partially complete, for example only the first 2 minutes of a 10 minute video working.

Would it be possible to get a setting so that if a download fails in that manner it deletes the partially downloaded file?

Thanks.

**Comment by 01000011-shade** [comment]
@alexjshaw It's not nearly as neat as a setting button to allow this, but addressing the same issue over in #34, you may wish to adjust your downloader.py file in the lines below to remove the file immediately after encountering a failure. As noted on the issue, it does prevent retries on files that may have failed for other reasons (these would get picked up on a re-run), but it will prevent you from having half baked files in your folder while the team works on a setting/retry threshold.

https://github.com/Emy69/CoomerDL/blob/f44f77de6cdad603f91277f870ed61e3bb45bbd9/downloader/downloader.py#L253-L255

Change to
```
        except Exception as e:
            self.log(self.tr("Error downloading: {e}", e=e))
            os.remove(filepath)
```

You may actually be able to get away with removing the file with `os.remove` and still appending the `media_url` to the `failed_files` list, but it is a known issue that retries on the `failed_files` list potentially puts you in an infinite loop of retries, particularly for files failing with the IncompleteRead Exception.

**Reply by alexjshaw** [reply]
Thanks, but I don't quite know how to do that.
I'm running the software from the zip download which contains an exe and only contains a compiled version of the file you referenced (as downloader.pyc), not the source code version which contains downloader.py
Is there any way to make the change on the compiled version?
I'll be honest, I know literally nothing about python so can't even figure out how to run the source code version!
